src/frequenz/sdk/config/_manager.py

```diff
@@ -150,7 +150,12 @@ async def new_receiver(  # pylint: disable=too-many-arguments # noqa: DOC502
         *,
         wait_for_first: bool = False,
         skip_unchanged: bool = True,
-        key: str | None = None,
+        # This is tricky, because a str is also a Sequence[str], if we would use only
+        # Sequence[str], then a regular string would also be accepted and taken as
+        # a sequence, like "key" -> ["k", "e", "y"]. We should never remove the str from
+        # the allowed types without changing Sequence[str] to something more specific,
+        # like list[str] or tuple[str].
+        key: str | Sequence[str] | None = None,
```

Comment:
sorry, This won't work....

Reply:
I didn't check if it works with toml files...

Reply:
I thought of accepting dot notation, but thought it might be better to keep the flexibility of allowing dots in the keys, which is supported by TOML. Also since in most cases we'll just use `"key"` or `[key, "sub_key"]`, or even `key.append(sub_key)`, as we probably want to do a chain for sub-keys, it sounds like at the end it might be more convenient to use sequence than the dot notation, which would mean something like `f"{key}.sub_key"`.

In any case, `ConfigParser` is not an option as it uses some weird INI file format, not TOML, and we decided to go with TOML for config files. `ConfigParser` is a very old beast, with a non-standard format, so I would rather avoid it for something new.

Reply:
I tried simplifying this but Python really sucks in terms of sequences of strings.

* I tried accepting only `Sequence[str]` but it doesn't work because a `str` is also a `Sequence[str]`, so it will keep accepting both, we'll just interpret a `str` as a sequence of characters, which is not what we want.
* I tried accepting only `tuple[str]`, but it sucks that a tuple with only one element needs a trailing comma, which is another common pitfall in Python (`"str" == ("str")`). Also `tuple("str") == ("s", "t", "r")` because "str" is a `Sequence[str]` :roll_eyes: 
* I tried accepting only `list[str]` but then we can't use an explicit default, as lists are mutable, and we hit just one more common Python pitfall. For example with this function:

  ```py
  def f(l: list[str] = ["logging"]) -> None:
      l.append("this sucks")
      print(l)`
  f()
 ```

  Prints:
  ```
  ['logging', 'this sucks']
  ```

`python -m this`
> There should be one-- and preferably only one --obvious way to do it.
> Although that way may not be obvious at first unless you're Dutch.

I guess the problem is I'm not Dutch.

----

With this in mind, maybe accepting dot notation with plain strings might be worth considering, but I'm also a bit hesitant about having to parse these strings, and parsing back ticks, my feeling is we could get into another problem and then have subtle bugs if there is an unmatched tick or stuff like that.

So I will keep it as is for now and we can reconsider it in a near future. If we go that route, I think I would just assume some restrictions on valid keys. We already have restrictions when creating the config files,I think only ASCII letters `[A-Z]` and digits `[0-9]` plus underscore `_` and dot `.` are accepted. I just didn't want to have this limitation hardcoded in the SDK for now, as maybe some user would want to write their config files manually and not have these restrictions.

Reply:
Ok, lets keep it as it is for now.
Wow such simple thing turns out to be so complicated...